app/src/main/java/com/example/hungrywolfs/model/DetailsViewModel.kt

```diff
@@ -15,17 +16,28 @@ class DetailsViewModel : ViewModel() {
     private var _mealDetails = MutableLiveData<MealDetailsInfo>()
     val mealDetails: LiveData<MealDetailsInfo> = _mealDetails
 
+    private var _favoriteMeals = MutableLiveData<MutableList<MealDetailsInfo>>()
+
     val mealTags: LiveData<List<String>> = _mealDetails.map {
         it.strTags?.split(",") ?: listOf()
     }
 
+    val isFavorite = MutableLiveData(false)
+
+    init {
+        _favoriteMeals.value = Hawk.get<MutableList<MealDetailsInfo>>("favorites")
+    }
+
     fun getMealWithDetails(idMeal: String) {
         viewModelScope.launch {
             try {
                 _mealDetails.value =
                     MealApi.retrofitService.getMealDetails(idMeal).meals.getOrNull(0)
+                if (_favoriteMeals.value?.contains(_mealDetails.value) == true) {
+                    isFavorite.value = true
```

Comment:
You can reduce all of this to `isFavorite.value = _favoriteMeals.value?.contains(_mealDetails.value)`
Technically you assign true when other value is true, so the check is not necessary